test/system/255-auto-update.bats

```diff
@@ -0,0 +1,243 @@
+#!/usr/bin/env bats   -*- bats -*-
+#
+# Tests for automatically update images for containerized services
+#
+
+load helpers
+
+UNIT_DIR="/usr/lib/systemd/system"
+SNAME_FILE=$BATS_TMPDIR/services
+
+function setup() {
+    skip_if_remote "systemd tests are meaningless over remote"
+    skip_if_rootless
+
+    basic_setup
+}
+
+function teardown() {
+    while read line; do
+        if [[ "$line" =~ "podman-auto-update" ]]; then
+            echo "Stop timer: $line.timer"
+            systemctl stop $line.timer
+            systemctl disable $line.timer
+        else
+            systemctl stop $line
+        fi
+        rm -f $UNIT_DIR/$line.{service,timer}
+    done < $SNAME_FILE
+
+    rm -f $SNAME_FILE
+    basic_teardown
+}
+
+function generate_service() {
+    fake_img=$1
+    target_img=$2
+    cmd=$3
+    label=$4
+
+    cname=$(random_string)
+    echo $cname
+    run_podman tag $fake_img $target_img
+    if [[ -n "$label" ]]; then
+        label="--label io.containers.autoupdate=$label"
+    fi
+    run_podman run -d --name $cname $label $target_img $cmd
+
+    run_podman generate systemd --new $cname
+    echo "$output" > "$UNIT_DIR/container-$cname.service"
+    echo "container-$cname" >> $SNAME_FILE
+    run_podman rm -f $cname
+
+    systemctl daemon-reload
+
+    run systemctl start container-$cname
+    if [ $status -ne 0 ]; then
+        die "Error starting systemd unit $SERVICE_NAME, output: $output"
+    fi
+
+    run systemctl status container-$cname
+    if [ $status -ne 0 ]; then
+        die "Non-zero status of systemd unit $SERVICE_NAME, output: $output"
+    fi
+}
+
+# This test can fail in dev. environment because of SELinux.
+# quick fix: chcon -t container_runtime_exec_t ./bin/podman
+@test "podman auto-update - label io.containers.autoupdate=image" {
+    run_podman images
+    run generate_service $IMAGE quay.io/libpod/alpine:latest top image
+    cname=${lines[0]}
+    echo $output
+    run_podman inspect --format "{{.Image}}" $cname
+    ori_image=$output
+
+    sleep 2
+    run_podman ps -a
+    run_podman auto-update
+    is "$output" "Trying to pull.*" "Image is updated."
+    run_podman ps -a
+    sleep 2
+    run_podman inspect --format "{{.Image}}" $cname
+    [[ "$output" != "$ori_image" ]]
+}
```

Comment:
Tests should clean up after themselves: please `run_podman rmi $UPDATE_IMAGE` here and in all tests, where `UPDATE_IMAGE` (or variable of your choice) is defined at the top as a way to not hardcode `quay.io/libpod/alpine:latest` everywhere.